### PR TITLE
Dynamically generate color picker background.

### PR DESCRIPTION
**Pull request by u-train**
This allows us to dynamically show the expected color based on value. Consequently, it's no longer an image with a limited resolution.

<details>
<summary> Funky cool images </summary>

image
image

</details>

### CODE REVIEW

**Comment by sircfenner** [comment]
Couple of points here-

1. Do we actually want this behavior where the hue+sat selector changes based on value? I note that the win10 color picker and a few others I could find do not do this. Not sure about it myself. A library consumer may want to preview the selected color in their own specific way. Since this is roughly modelled on the win10 picker right now I would probably prefer to leave this change out

2. New hue+sat pane generation is good but currently the hue/sat gradiented subframes are representing color incorrectly because the uigradient is composing over their default backgroundcolor3, which is a medium gray. Suggested a fix

File: src/ColorPicker/init.lua [hunk 1]
```diff
@@ -103,6 +113,24 @@ function ColorPicker:render()
 						BackgroundColor3 = Color3.fromRGB(0, 0, 0),
 					}),
 				}),
+				HueGradient = Roact.createElement("Frame", {
```
Comment (sircfenner) [inline src/ColorPicker/init.lua:116]:
```suggestion
				HueGradient = Roact.createElement("Frame", {
					BackgroundColor3 = Color3.fromRGB(255, 255, 255),
```

File: src/ColorPicker/init.lua [hunk 2]
```diff
@@ -103,6 +113,24 @@ function ColorPicker:render()
...
+					Size = UDim2.fromScale(1, 1),
+					ZIndex = -1
+				}, {
+					Gradient = Roact.createElement("UIGradient", {
+						Color = generateHueKeypoints(val),
+					}),
+				}),
+				SaturationGradient = Roact.createElement("Frame", {
```
Comment (sircfenner) [inline src/ColorPicker/init.lua:124]:
```suggestion
				SaturationGradient = Roact.createElement("Frame", {
					BackgroundColor3 = Color3.fromRGB(255, 255, 255),
```

Comment (sircfenner) [inline src/ColorPicker/init.lua:124]:
```suggestion
				SaturationGradient = Roact.createElement("Frame", {
					BackgroundColor3 = Color3.fromRGB(255, 255, 255),
```

**Comment by u-train** [comment]
> A library consumer may want to preview the selected color in their own specific way.
This change doesn't stop them from previewing the selected color at all. This change

**Comment by sircfenner** [comment]
Looking good, thanks. Just one more change needed so that the cross/indicator on the sat-hue panel is visible for low value components. Currently it becomes invisible due to the low contrast between black and very dark/black colors. 

GitHub PR doesn't allow commenting on unchanged lines so I can't propose my suggested changes as part of this review. If you enable "Allow edits from maintainers" on your PR I can directly commit my changes, otherwise see below:

```diff
local hue, sat, val = props.Color:ToHSV()

+local indicatorColor = Color3.fromRGB(0, 0, 0)
+if val < 0.4 then
+	indicatorColor = Color3.fromRGB(200, 200, 200)
+end
```

Then for each of Vertical and Horizontal, under Indicator:
```diff
-BackgroundColor3 = Color3.fromRGB(0, 0, 0),
+BackgroundColor3 = indicatorColor,
```

**Comment by sircfenner** [comment]
Perfect, thanks again.

**Pull request merged**